src/service/rest/result.rs

```diff
@@ -0,0 +1,80 @@
+use diesel;
+use diesel::result::{Error as DieselError, QueryResult};
+use rocket::http::Status;
+use rocket::request::Request;
+use rocket::response::{Responder, Response};
+use rocket_contrib::Json;
+use serde::{Serialize, Serializer};
+use serde::ser::SerializeMap;
+use serde_json;
+use std::convert::From;
+use std::io::Cursor;
+
+#[derive(Debug)]
+pub struct ApiError {
+    http_status: u16,
+    message: String,
+}
+
+impl Serialize for ApiError {
```

Comment:
You should be able to derive this.

Reply:
It has a different shape than the default `derive`. I wanted to remove the `http_status` field, since that is already used to create the response status code. I also wanted to add the `"status": "error"`field to match the current Python APi.